### PR TITLE
Random code fixes and improvements

### PR DESCRIPTION
**Pull request by Egorand**
- Mostly small code style issues fixed by auto-formatting
- Replaced a number of statements to use more Kotlin features, where it makes sense

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/FunSpec.kt [hunk 1]
```diff
@@ -55,12 +55,8 @@ class FunSpec private constructor(builder: Builder) {
     }
   }
 
-  internal fun parameter(name: String): ParameterSpec? {
-    for (parameter in parameters) {
-      if (parameter.name == name) return parameter
-    }
-    return null
-  }
+  internal fun parameter(name: String): ParameterSpec? =
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/FunSpec.kt]: Return type can be omitted here

Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/FunSpec.kt]: Thought about keeping it to make it more clear that it's nullable, but it's probably clear anyway

File: src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt [hunk 2]
```diff
@@ -56,14 +54,9 @@ class FunSpecTest {
...
-    for (executableElement in elements) {
-      if (executableElement.simpleName.toString() == name) {
-        return executableElement
-      }
-    }
-    throw IllegalArgumentException(name + " not found in " + elements)
-  }
+  private fun findFirst(elements: Collection<ExecutableElement>, name: String) =
+      elements.firstOrNull { it.simpleName.toString() == name } ?:
+          throw IllegalArgumentException(name + " not found in " + elements)
```
Comment (NightlyNexus) [inline src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: could also be `requireNotNull(elements.firstOrNull { it.simpleName.toString() == name }) { name + " not found in " + elements }`

Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: Yep, good one! I guess I will submit another small PR

**Pull request merged**